garage/tf/algos/vpg.py

```diff
@@ -87,37 +86,52 @@ def init_opt(self):
             else:
                 valid_var = None
 
-            dist_info_vars = self.policy.dist_info_sym(obs_var,
-                                                       state_info_vars)
-            logli = dist.log_likelihood_sym(action_var, dist_info_vars)
-            kl = dist.kl_sym(old_dist_info_vars, dist_info_vars)
-
-            # formulate as a minimization problem
-            # The gradient of the surrogate objective is the policy gradient
-            if is_recurrent:
+        dist_info_vars = self.policy.dist_info_sym(obs_var, state_info_vars)
+        logli = dist.log_likelihood_sym(action_var, dist_info_vars)
+        kl = dist.kl_sym(old_dist_info_vars, dist_info_vars)
+        surr_obj_scope = tf.name_scope(
+            "surr_obj", values=[logli, advantage_var, valid_var])
+        mean_kl_scope = tf.name_scope("mean_kl", values=[kl, valid_var])
```

Comment:
same comment about splitting the block to eliminate saving scope variables (which is difficult to read)